src/api/app/controllers/webui/users/block_controller.rb

```diff
@@ -0,0 +1,41 @@
+class Webui::Users::BlockController < Webui::WebuiController
+  before_action :check_displayed_user, only: %i[create destroy]
+  before_action :set_user_block, only: :destroy
+
+  after_action :verify_authorized
+
+  def create
+    @user_block = User.session!.blocked_users.new(blocked: @displayed_user)
+
+    authorize @user_block
+
+    if @user_block.save
+      flash[:success] = "User '#{@user_block.blocked.login}' was successfully blocked."
+    else
+      flash[:error] = "Failed to block user: #{@user_block.errors.full_messages.to_sentence}."
+    end
+
+    redirect_back_or_to user_path(@displayed_user)
+  end
+
+  def destroy
+    authorize @user_block
+
+    if @user_block.destroy
+      flash[:success] = "User '#{@user_block.blocked.login}' was successfully unblocked."
+    else
+      flash[:error] = "Failed to unblock user: #{@user_block.errors.full_messages.to_sentence}"
+    end
+
+    redirect_back_or_to user_path(@displayed_user)
```

Comment:
Looks like the redirection does not take into account anchors, if I'm in the comments tab of the user's profile, and I unblock it, I go back to the first tab of the user profile, not to the comments tab...